### PR TITLE
Split LRU list

### PR DESCRIPTION
**Pull request by arutk**

### CODE REVIEW

File: src/eviction/lru.c [hunk 1]
```diff
@@ -163,48 +163,121 @@ void evp_lru_init_cline(ocf_cache_t cache, ocf_cache_line_t cline)
...
-	list = metadata_test_dirty(cache, cline) ?
-		&part->runtime->eviction.policy.lru.dirty :
-		&part->runtime->eviction.policy.lru.clean;
-
+	list = evp_get_list(cache, cline);
 	remove_lru_list(cache, list, cline);
 }
 
+static inline void lru_iter_init(ocf_cache_t cache, ocf_part_id_t part_id,
+	uint32_t start_evp, bool clean, struct ocf_lru_iter *iter)
```
Comment (robertbaldyga) [inline src/eviction/lru.c]: As a convention constructor takes pointer to initialized object as a first parameter.

File: src/eviction/lru.c [hunk 2]
```diff
@@ -395,43 +473,48 @@ void evp_lru_init_evp(ocf_cache_t cache, ocf_part_id_t part_id)
...
 {
 	struct ocf_user_part *part = &cache->user_parts[part_id];
+	uint32_t ev_list = (cline % OCF_NUM_EVICTION_LISTS);
 	struct ocf_lru_list *clean_list;
 	struct ocf_lru_list *dirty_list;
 
-	clean_list = &part->runtime->eviction.policy.lru.clean;
-	dirty_list = &part->runtime->eviction.policy.lru.dirty;
+	clean_list = &part->runtime->eviction[ev_list].policy.lru.clean;
+	dirty_list = &part->runtime->eviction[ev_list].policy.lru.dirty;
```
Comment (robertbaldyga) [inline src/eviction/lru.c:488]: `evp_get_list()` could be used instead.

File: src/eviction/lru.c [hunk 3]
```diff
@@ -395,43 +473,48 @@ void evp_lru_init_evp(ocf_cache_t cache, ocf_part_id_t part_id)
...
```
Comment (robertbaldyga) [inline src/eviction/lru.c:504]: `evp_get_list()` could be used instead.

**Review by robertbaldyga — CHANGES_REQUESTED** [review]

File: src/metadata/metadata_partition_structs.h [hunk 4]
```diff
@@ -56,6 +54,8 @@ struct ocf_lru_iter {
 struct ocf_user_part {
 	struct ocf_user_part_config *config;
 	struct ocf_user_part_runtime *runtime;
+	struct ocf_refcnt cleaning;
+	unsigned id;
```
Comment (mmichal10) [inline src/metadata/metadata_partition_structs.h:58]: You can use `ocf_part_id_t` instead of unsigned

**Review by robertbaldyga — APPROVED** [review]

**Review by Ostrokrzew — APPROVED** [review]

**Review by mmichal10 — APPROVED** [review]

**Review by Ostrokrzew — APPROVED** [review]

**Pull request merged**